vignettes/decorate-modules-output.Rmd

```diff
@@ -0,0 +1,389 @@
+
+```{r}
+static_decorator_lang <- teal_transform_module(
+  server = make_teal_transform_server(
+    expression(
+      plot <- plot +
+        ggtitle("This is title") +
+        xlab("x axis title")
+    )
+  )
+)
+```
+
+### UI
+
+To create a decorator with user interactivity, you can incorporate a `shiny` UI module. 
+In the example below, the x-axis title is set dynamically via a `textInput`, allowing users to specify their preferred label. 
+Note how the input parameters are passed to the `within` function (e.g., `x_axis_title`):
+
+```{r}
+interactive_decorator <- teal_transform_module(
+  ui = function(id) {
+    ns <- NS(id)
+    div(
+      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+    )
+  },
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive({
+        req(data())
+        within(data(),
+          {
+            plot <- plot +
+              ggtitle("This is title") +
+              xlab(x_axis_title)
+          },
+          x_axis_title = input$x_axis_title
+        )
+      })
+    })
+  }
+)
+```
+
+As in the earlier examples, `make_teal_transform_server(expr)` can simplify the creation of the server component. 
+This wrapper allows you to use `input` object names directly in the expression:
+
+```{r}
+interactive_decorator_lang <- teal_transform_module(
+  ui = function(id) {
+    ns <- NS(id)
+    div(
+      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+    )
+  },
+  server = make_teal_transform_server(
+    expression(
+      plot <- plot +
+        ggtitle("This is title") +
+        xlab(x_axis_title)
+    )
+  )
+)
+```
+
+## Handling Various Object Names
+
+`teal_transform_module` relies on the names of objects created within a module. 
+Writing a decorator that applies to any module can be challenging because different modules may use different object names.
+It is recommended to create a library of decorator functions that can be adapted to the specific object names used in `teal` modules.
+In the following example, focus on the `output_name` parameter to see how decorator can be applied to multiple modules:
+
+```{r}
+gg_xlab_decorator <- function(output_name) {
+  teal_transform_module(
+    ui = function(id) {
+      ns <- NS(id)
+      div(
+        textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+      )
+    },
+    server = function(id, data) {
+      moduleServer(id, function(input, output, session) {
+        reactive({
+          req(data())
+          within(data(),
+            {
+              output_name <- output_name +
+                xlab(x_axis_title)
+            },
+            x_axis_title = input$x_axis_title,
+            output_name = as.name(output_name)
+          )
+        })
+      })
+    }
+  )
+}
+```
+
+Decorator failures are managed by an internal `teal` mechanism called `trigger_on_success`, which ensures that the `data`
```

Comment:
Original function name is `.trigger_on_success` but I would try to avoid giving the internal name - to follow the rule to never expose any internals in the public documentation
 ```suggestion
Decorator failures are managed by an internal `teal` mechanism called "trigger on success", which ensures that the `data`
```

Reply:
f9bdb60c692ffc465975f747232eee164b36a2d9